.github/actions/preload-img-cache/action.yml

```diff
@@ -0,0 +1,47 @@
+name: Preload img cache
+inputs:
+  device:
+    description: 'Device name'
+    required: true
+  filename:
+    description: 'Name of file in workdir'
+    required: true
+  test-db-path:
+    description: 'Path to test database'
+    required: true
+outputs:
+  cache-hit:
+    description: 'Did we hit the cache?'
+    value: ${{ steps.cache-img.outputs.cache-hit }}
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/cache@v3
+      id: cache-img
+      with:
+        key: img-${{ inputs.device }}-${{ inputs.filename }}
+        path: |
+          workdir/${{ inputs.filename }}
+    - if: ${{ ! steps.cache-img.outputs.cache-hit }}
+      uses: awalsh128/cache-apt-pkgs-action@v1
+      with:
+        packages: python3-lz4 python3-protobuf
+    - if: ${{ ! steps.cache-img.outputs.cache-hit }}
+      uses: awalsh128/cache-apt-pkgs-action@v1
+      with:
+        packages: python3-tomlkit
+    - name: Creating workdir
+      if: ${{ ! steps.cache-img.outputs.cache-hit }}
+      shell: sh
+      run: mkdir -p workdir
+    - name: Downloading device image for ${{ inputs.device }}
+      if: ${{ ! steps.cache-img.outputs.cache-hit }}
+      shell: sh
+      run: |
+        ./tests_ci/dummy_image.py \
+          download \
+          --no-compress \
```

Comment:
That is indeed why I added that feature into `dummy_image.py`, but I found out that the cache action also compresses the image while caching so it would be a bit redundant. If the test script would be changed to also allow compressed images as well (decompressing them on-the-fly), it would make sense to compress them while caching to save disk usage when in use.

Reply:
Ahh, I see. I don't think we can have the test script use the compressed file directly, because avbroot depends on random access to the input file.

We might be able to have the dummy file downloader create sparse files though, to save on disk space. It would just need to seek() past the holes without actually writing `\0`.